src/js/webrequest.js

```diff
@@ -706,6 +715,7 @@ function startListeners() {
   chrome.webRequest.onHeadersReceived.addListener(onHeadersReceived, {urls: ["<all_urls>"]}, ["responseHeaders", "blocking"]);
   chrome.tabs.onRemoved.addListener(onTabRemoved);
   chrome.tabs.onReplaced.addListener(onTabReplaced);
+  chrome.tabs.onUpdated.addListener(onTabUpdated);
```

Comment:
Possibly, if that makes sense. Registering twice for the same thing seems like extra overhead although how much (a trivial amount?) is unclear. It might also make sense to split off the badge changes into their own PR.